MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp

```diff
@@ -73,11 +73,10 @@ void QtReflMainView::initLayout() {
 * @param menu : [input] The menu where actions will be added
 * @param command : [input] The command (action) to add
 */
-void QtReflMainView::addToMenu(QMenu *menu, ReflCommandBase_uptr command) {
+void QtReflMainView::addToMenu(QMenu *menu, ReflCommand_uptr command) {
 
-  // cppcheck-suppress unreadVariable
-  ReflCommandAdapter *adapter =
-      new ReflCommandAdapter(menu, std::move(command));
+  m_adapters.push_back(Mantid::Kernel::make_unique<ReflCommandAdapter>(
```

Comment:
You should purge your `m_adapters` at appropriate times if you're not doing so already.